src/main/kotlin/com/healthmetrix/labres/lab/UpdateResultUseCase.kt

```diff
@@ -26,18 +26,24 @@ class UpdateResultUseCase(
         }
 
         val sample = updateResultRequest.type.sample
-        val orderInfo = repository.findByOrderNumberAndSample(orderNumber, sample)
-            ?: return UpdateResult.ORDER_NOT_FOUND
-
-        val update = orderInfo.copy(
-            status = updateResultRequest.result.asStatus(),
-            labId = labId,
-            testType = updateResultRequest.type
-        )
-
-        updateTimestamp(update, updateResultRequest.result, now)
-            .let(repository::save)
-            .also { notifyOnStatusChange(it.id, it.notificationUrls) }
+        val existingOrders = repository.findByOrderNumberAndSample(orderNumber, sample)
+
+        if (existingOrders.isEmpty())
+            return UpdateResult.ORDER_NOT_FOUND
+
+        val ordersToUpdate = existingOrders.map { existing ->
```

Comment:
not a blocker but it seems like the logic in the 2nd `.map` can be included in the first `.map` no?